crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2957,6 +2962,54 @@ impl Room {
             .await?;
         Ok(())
     }
+
+    /// Remove verification requirements for the given users and
+    /// resend messages that failed to send because their identities were no
+    /// longer verified (in response to
+    /// `SessionRecipientCollectionError::VerifiedUserChangedIdentity`)
+    #[cfg(feature = "e2e-encryption")]
+    pub async fn withdraw_verification_and_resend(
+        &self,
+        user_ids: Vec<OwnedUserId>,
+        transaction_id: &TransactionId,
+    ) -> Result<()> {
+        let encryption = self.client().encryption();
+
+        for user_id in user_ids {
+            if let Some(user_identity) = encryption.get_user_identity(&user_id).await? {
+                user_identity.withdraw_verification().await?;
+            }
+        }
+
+        self.send_queue().unwedge(transaction_id).await?;
```

Comment:
The API here is weird, because it's undocumented anywhere what the transaction_id is for, and that's it's one managed by the send queue. I'd remove this line from this function and the other, and instead have the callers at the FFI layer call the `unwedge` function and document the transaction id properties.

Reply:
Just to make sure I get it straight: 
1) We would move `withdraw_verification` and `trust_devices` to Encryption
2) We would expose unwedge by txnId on the Room level
3) The final client would first trust via Client and then resend via Room

Does that sounds about right?

Reply:
> We would expose unwedge by txnId on the Room level

Do we need to, or can the caller themselves do `room.send_queue().unwedge(transaction_id)`? The idea here is that we should make it clear why it's ok to pass a transcation id to the send queue; it's because it's originating from the send queue too, and this information is super hidden at the moment, with the current API shape.

To be honest, I'm thinking all of this, the loop included above, could happen at the FFI layer, if it's just calling into other existing APIs without adding other value.

Consider the semantics carefully though: if any of the verification withdrawing fails, is it ok that 1. others have been successful, 2. we don't unwedge in that case?

Reply:
> Do we need to, or can the caller themselves do room.send_queue().unwedge(transaction_id)

Oh I see, I'm thinking of it from the FFI layer and you're talking SDK. Yeah that would be totally okay

> all of this, the loop included above, could happen at the FFI layer

Right yeah, I agree!

> Consider the semantics carefully though

I have, I think that bit's fine. Retrying to send it while some of the devices/users from the original failure haven't been resolved will result in the same error.